### PR TITLE
track add-a-question, click-save actions

### PR DESCRIPTION
**Pull request by sravfeyn**

### CODE REVIEW

File: src/core.js [hunk 1]
```diff
@@ -155,6 +155,9 @@ define([
                 }
                 _this.ensureCurrentMugIsSaved(function () {
                     _this.validateAndSaveXForm(forceFullSave);
+                    if (typeof window.analytics !== "undefined") {
```
Comment (emord) [inline src/core.js]: Should probably use !_.isUndefined(window.analytics)

File: src/core.js [hunk 2]
```diff
@@ -1406,6 +1409,10 @@ define([
                 $firstInput.focus().select();
             }
         });
+
+        if (typeof window.analytics !== "undefined") {
+            window.analytics.track("Added a question");
+        }
```
Comment (millerdev) [inline src/core.js]: This should be moved up before line 1406

File: src/core.js [hunk 3]
```diff
@@ -155,6 +155,9 @@ define([
                 }
                 _this.ensureCurrentMugIsSaved(function () {
                     _this.validateAndSaveXForm(forceFullSave);
+                    if (typeof window.analytics !== "undefined") {
+                        window.analytics.track("Clicked Save in Formbuilder");
+                    }
```
Comment (millerdev) [inline src/core.js]: I would move this above the call to `_this.validateAndSaveXForm` in case it raised an exception.

**Comment by emord** [comment]
Just ran into this: https://github.com/dimagi/Vellum/blob/b06a065f06160e11427e2368047158713460989a/src/widgets.js#L219

Should that be changed to use windows.analytics.track?

**Comment by sravfeyn** [comment]
Those are for Google Analytics this PR is related to Segment.com analytics.

**Comment by millerdev** [comment]
I'll push a new version of Vellum to HQ today so this gets deployed.

**Comment by millerdev** [comment]
pushed to HQ. should go live on the next deploy

**Comment by sravfeyn** [comment]
Thanks Daniel!

**Pull request merged**